kernel/user.c

```diff
@@ -61,7 +68,7 @@ int user_read_string(addr_t addr, char *buf, size_t max) {
     read_wrlock(&current->mem->lock);
     size_t i = 0;
     while (i < max) {
-        if (__user_read_task(current, addr + i, &buf[i], sizeof(buf[i]))) {
+        if (__user_read_task(current, addr + i, &buf[i], sizeof(buf[i])), false) {
```

Comment:
uh does this compile? no parameter was added to __user_read_task

Reply:
You're reading it wrong, this is using the comma operator

Reply:
this seems like a bad idea, if you're going to ignore the return value, why put the call in the if at all? very easy to read wrong as you've seen

Reply:
I think you're still reading it wrong